### PR TITLE
Web sockets support

### PR DESCRIPTION
**Pull request by serkantkaraca**
Adding web sockets support for AMQP. This change depends on the next Nuget release of Azure.Amqp. Once package is released we can check this in.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs [hunk 1]
```diff
@@ -7,6 +7,22 @@ namespace Microsoft.Azure.EventHubs
     using System.Text;
 
     /// <summary>
+    ///  Supported transport types
+    /// </summary>
+    public enum TransportTypes
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs]: TransportType

File: src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs [hunk 2]
```diff
@@ -7,6 +7,22 @@ namespace Microsoft.Azure.EventHubs
...
+    {
+        /// <summary>
+        /// Transport type AMQP
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs]: AMQP over the default TCP transport protocol.

File: src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs [hunk 3]
```diff
@@ -7,6 +7,22 @@ namespace Microsoft.Azure.EventHubs
...
+        /// </summary>
+        Amqp,
+
+        /// <summary>
+        /// Transport type AMQP over web sockets
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs]: AMQP over the Web Sockets transport protocol.

File: src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs [hunk 4]
```diff
@@ -144,6 +161,13 @@ public EventHubsConnectionStringBuilder(string connectionString)
         public TimeSpan OperationTimeout { get; set; }
 
         /// <summary>
+        /// Transport type for the client connection.
+        /// Avaiable options are Amqp and AmqpWebSockets.
+        /// Defaults to Amqp if not specified.
+        /// </summary>
+        public TransportTypes? TransportType { get; set; }
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs]: It does not need to be nullable. Just set the default value in the constructor.

File: src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs [hunk 5]
```diff
@@ -112,6 +114,20 @@ public class EventHubsConnectionStringBuilder
             string entityPath,
             string sharedAccessSignature,
             TimeSpan operationTimeout)
+            : this(endpointAddress, entityPath, operationTimeout)
+        {
+            if (string.IsNullOrWhiteSpace(SharedAccessSignature))
```
Comment (alexsorokoletov) [inline src/Microsoft.Azure.EventHubs/Primitives/EventHubsConnectionStringBuilder.cs]: This line produces ArgumentNullException for anyone who calls this constructor.

**Comment by alexsorokoletov** [comment]
This PR made `EventHubsConnectionStringBuilder` ctor unusable when supplying SharedAccessSignature.

**Pull request merged**